### PR TITLE
 Added implement a registration service and wrote unit tests for it. 

### PR DESCRIPTION
**Pull request by SofiiaKazanivska**
there was an error with threads and requests, I didn't know how to fix it, so I created a new repository and deleted the previous one

### CODE REVIEW

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -5,10 +5,32 @@
...
+        }
+
+        if (user.getLogin() == null) {
+            throw new RuntimeException("The login can not be null!");
+        }
+
+        if (user.getPassword() == null) {
+            throw new RuntimeException("The password can not be null!");
+
+        }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:29]: Can you check `user.getPassword().length()` before `user.getPassword() == null` ?

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        alice.setPassword("123Alice");
+        alice.setAge(18);
+
+        bob.setLogin("Bob");
+        bob.setPassword("123Bob");
+        bob.setAge(22);
+    }
+
+    @Test
+    void register_age_less_than_18_notOk() {
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:35]:
```suggestion
    void register_ageLessThan18_notOk() {
```
same logic for other methods names

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    @Test
+    void register_nullAge_notOk() {
+        bob.setAge(null);
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(bob);
+        });
+    }
+
+    @Test
+    void register_the_same_login_notOk() {
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:64]:
```suggestion
    void register_sameLogin_notOk() {
```
do not use articles in the names

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java (hunk 1 above)
Comment (SofiiaKazanivska) [reply src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:29]: What are you mean? i did it before ( user.getPassword() == null is before user.getPassword().length() )

**Pull request closed without merging**